app/Services/Notification/MarkReadService.php

```diff
@@ -17,15 +18,15 @@ public function __invoke(Student $student, int $notificationId): void
     {
         $notification = Notification::where('id', $notificationId)
             ->where('type', TypeEnum::ONCE)
-            ->with(['students', 'course'])
+            ->with(['students'])
             ->firstOrFail();
+        
+        // 該当生徒の受講期限(attendance_deadline)を取得
+        $deadline = Attendance::where('student_id', $student->id)
+            ->where('course_id', $notification->course_id)
+            ->value('attendance_deadline');
 
-        if (
-            $notification->course->attendance_deadline &&
-            CarbonImmutable::now()->gte(
-                $notification->course->attendance_deadline->endOfDay()
-            )
-        ) {
+        if ($deadline && CarbonImmutable::now()->gte(CarbonImmutable::parse($deadline)->endOfDay())) {
```

Comment:
nullではない判定は明示的に記載するようにしましょう！
PHPはこの判定条件でバグの温床になるためでした...。

```suggestion
        if ($deadline !== null && CarbonImmutable::now()->gte(CarbonImmutable::parse($deadline)->endOfDay())) {
```